### PR TITLE
Deprecating The Datadog Processor

### PR DESCRIPTION
**Pull request by gord02**
The Datadog Processor is deprecated in favor of the Datadog Connector

Link to issue: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/19740

### CODE REVIEW

File: processor/datadogprocessor/processor.go [hunk 1]
```diff
@@ -51,6 +51,7 @@ func newProcessor(ctx context.Context, logger *zap.Logger, config component.Conf
 	if err != nil {
 		return nil, err
 	}
+	logger.Info("This component is deprecated in favor of the Datadog connector")
```
Comment (dineshg13) [inline processor/datadogprocessor/processor.go:54]: Add as warning

File: processor/datadogprocessor/README.md [hunk 2]
```diff
@@ -2,17 +2,20 @@
...
 
-[beta]: https://github.com/open-telemetry/opentelemetry-collector#beta
+[deprecated]: https://github.com/open-telemetry/opentelemetry-collector#deprecated
 [contrib]: https://github.com/open-telemetry/opentelemetry-collector-releases/tree/main/distributions/otelcol-contrib
 <!-- end autogenerated section -->
 
 ## Description
 
+
+This component has been deprecated in favor of its predecessor the [Datadog Connector](../../connector/datadogconnector/README.md) which serves the same purpose and should be used in place of the Datadog processor.
```
Comment (songy23) [inline processor/datadogprocessor/README.md:17]:
```suggestion
This component has been deprecated in favor of its successor the [Datadog Connector](../../connector/datadogconnector/README.md) which serves the same purpose and should be used in place of the Datadog processor.
```

**Review by songy23 — APPROVED** [review]

**Review by codeboten — COMMENTED** [review]
Please add a deprecation notice in the go.mod file

**Comment by songy23** [comment]
> Replace statements are out of date, please run "make crosslink" and commit the changes in this PR.

File: processor/datadogprocessor/go.mod [hunk 3]
```diff
@@ -1,3 +1,4 @@
+// Deprecated
```
Comment (codeboten) [inline processor/datadogprocessor/go.mod:1]:
```suggestion
// Deprecated: Datadog processor is deprecated in favor of Datadog connector
```

**Pull request merged**